src/main/java/de/komoot/photon/elasticsearch/Importer.java

```diff
@@ -34,8 +35,14 @@ public Importer(Client esClient, String languages) {
 	@Override
 	public void add(PhotonDoc doc) {
 		try {
-			this.bulkRequest.add(this.esClient.prepareIndex(indexName, indexType).
-					setSource(Utils.convert(doc, languages)).setId(String.valueOf(doc.getPlaceId())));
+                    XContentBuilder source = Utils.convert(doc, languages);
+                    long id = doc.getPlaceId();
+                    if(id > 0){
+                        this.bulkRequest.add(this.esClient.prepareIndex(indexName, indexType).setSource(source).setId(String.valueOf(id)));
+                    }
+                    else{
+                        this.bulkRequest.add(this.esClient.prepareIndex(indexName, indexType).setSource(source));
```

Comment:
that's not too nice: first you require the implicit knowledge that an id less than 1 is not a real id. second it may happen that an automatically created id will be overwritten by a nominatim id.

maybe it's just best to never set the id explicitly but let es do the job. we will loose the connection between a photon doc and a nominatim row, but this can be reestablished by using osm id and osm type (way, node, relation).